### ISSUE TITLE
Add API method to reject all without using UI

### ISSUE THREAD

**Issue by MathRobin** [InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework#198]
Hi !

Lot of people just don't want to be tracked, some of them have activated "Do not track" feature on their browsers (for those who know it), some of them use browsers extension to kick or manage permissions (I develop one of those extension).

But to develop it, and to be really "transparency - and - consent" you need to allow easily to reject all non-mandatory permissions. You already do it using UI. Good job, most of platforms don't provide it. So thanks ! But can you add this feature in the JS API ? 

Many thanks !

**Comment by alextcone** [comment]
@patrickverdon I'm thinking Filip may want to weigh in here from a policy perspective in terms of what TCF sets out to do and what it does not.

**Comment by filipsedefov** [comment]
Hi @MathRobin,
The TCF seeks to enable third party controllers to comply with GDPR requirements by providing detailed transparency about their identity/data processing purposes and allowing the user to make granular choices regarding those controllers and purposes. UIs are an essential element of those disclosures and choices. I’m afraid that adding an API method to reject all without surfacing a UI would defeat the purpose of the Framework, and hence its utility to both users and third party controllers.

**Comment by MathRobin** [comment]
But some people already that they just don't want any non mandatory cookies. Stats of my browser extension proved it. Without any massive advertisement, I got (and kept) 100 users on chrome, 100 on Firefox (mobile & desktop) the first month. So I can assume that I'm not alone in this case.

UIs is important when you want to chose site by site, yes, I agree with you. But when like me, you know that you don't want any cookies, never.

Moreover, you already provide API interface to accept everything and to reject one by one. So reject everything will not be a massive cost, considering that if you comply to Gdpr, it's the default choice. Quantcast, for example, provides this API method. Because they want to comply. Sites which implements Quantcast can choose if they accept to display or not the button for rejecting everything (hello hypocrisies).

**Comment by MathRobin** [comment]
And what about the respect of "do not track" feature/flag provided by some browsers. When I switch on "do not track" on my browser, do you still think that is important to ask me again and again on each site if I want to be tracked ?

**Comment by herrernst** [comment]
I also think a reject-all API function would be good to have!

**Comment by asr-enid** [comment]
`Moreover, you already provide API interface to accept everything and to reject one by one`

This might be a misunderstanding, the public CMP API is read only https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md to 

The other libraries you might be referring too are only use by CMPs to encode a status that established while interaction with a user via a UI (CMP) interface. https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/tree/master/Consent%20String%20SDK

**Comment by MathRobin** [comment]
OK so it's a misunderstanding for that point, you're true. I understand why you want to let user chose by UI.

But remember that, by default, the law ask that all consentis are rejected by default. To allow an explicit and chosen consent. And when, a user activates "do not track" feature of its browser, it knows what it does. So, again, why ask to those guys, if they want to be track if they have activated voluntarily the "do not track" feature.

**Comment by asr-enid** [comment]
`But remember that, by default, the law ask that all consentis are rejected by default. To allow an explicit and chosen consent.`

Absolutely - from a TCF perspective that is part of the policy CMP must adhere to: https://iabeurope.eu/iab-europe-transparency-consent-framework-policies/#C_Specific_Requirements_for_Framework_UIs_in_Connection_with_Requesting_a_Users_Consent

**Comment by MathRobin** [comment]
So why there is so much CMP platforms/sites which are using TCF CMP API and enables by default most of the consents when they should not ? It should not be customizable.
(I reuse my example of quantcast which allow sites to chose which consent is enabled by default or not or if user can reject all, **hi again to hypocrisis**)

Last but not least because it seems that this subject is always evaded, what about **Do Not Track** ? Again UI is and should be the standard way to define a choice, but, again, when user activates do not track, he already had chosen what it wants, so it's a non-sense to ask it again and again : it doesn't want to be tracked !

**Comment by MathRobin** [comment]
Hello! I'm back, are you still stuck in hypocrisis or do you care about we, the people, will be able to make some legal proceeds to all of your users ?
Quick reminder: https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/pull/254

Cheers

**Comment by MathRobin** [comment]
No news? Still don't want to respect law?

**Comment by dmdabbs** [comment]
> So why there is so much CMP platforms/sites which are using TCF CMP API and enables by default most of the consents when they should not ?

Hello @MathRobin. If you encounter a site that has what looks to you to be a TCF CMP installed and their UI has consent disclosures (not legitimate interest) enabled by default, then you should send notice of this to IAB Europe. The TCF operate a crawler-based compliance programme, but it cannot reach all sites. You can provide feedback to framework@iabeurope.eu. 
If it is associated with an authorised CMP, then TCF can take enforcement action. 
cc: @anderagakura

**Comment by MathRobin** [comment]
Ah ah. It's your job. Don't hide behind a "it's hard to follow". Your system is against law by design.

**Comment by anderagakura** [comment]
@MathRobin By default, the law asks that all consents are rejected by default. Today, we have a compliance program in order to crawl the sites and make sure the law is respected. But yes, we all know there is **a lot** of sites. 

Do not hesitate to send us your feedback to [framework@iabeurope.eu](mailto:framework@iabeurope.eu)

**Comment by MathRobin** [comment]
And why the hell there is no way to reject all in your API when there is the accept all? Is it still hypocrisies?

**Comment by anderagakura** [comment]
The core idea of TCF is to give the opportunity to the user to get the information and make choices. If there is no UI, an API method to reject everything, the TCF becomes useless for both the users and the third party controllers.

As mentioned before, our compliance program is a new layer to make sure the law is respected which is **all the consents are rejected by default**.

Today, this is the current status of the TCF and as each product, it (must) evolves over the time, with the community, with the users. Saying that, your feedback are definitely welcome to framework@iabeurope.eu.

**Comment by acautomacaoecontrole** [comment]
If "The core idea of TCF is to give the opportunity to the user to get the information and make choices", then simply add a "reject all" button and people will choose among reject all, accept all or do a customized allowance. It's really upsetting to be obligated to roll and revert all those infinite pre-allowed options. People do not have the time to waste cancelling the pre-allowed permissions nor they enjoy this so boring task every time they visit first time a website. Just add the "reject all" button to those (many!) who prefer to click it. No more discussion needed.

**Comment by MathRobin** [comment]
Normally you should ask again for new partners. It has been ask by european countries, theorically to protect consumers by seeing their privacy suc*ed by private companies. But as always, lobbies have done their job. The terms are "it should be as easy to reject as approving". So you can do whatever you want when displayed. Again, it's just hypocrisie to display "accept all" but not "reject all"
